src/auth/authContext.jsx

```diff
@@ -0,0 +1,62 @@
+import React, { useEffect, useState } from "react";
+import firestore, { auth } from '../firebaseConfig';
+
+export const AuthContext = React.createContext({
+  user: {},
+  setUser: () => { },
+});
+
+const AuthProvider = (props) => {
+  const [user, setUser] = useState({});
+
+  useEffect(() => {
+    auth.onAuthStateChanged((auth) => {
+      console.log('AUTHSTATECHANGE ', auth);
+      processAuth(auth,setUser);
+    });
+    return () => {
+      // cleanup
+    }
+  }, []);
+
+  return (
+    <AuthContext.Provider value={{ user, setUser }}>
+      {props.children}
+    </AuthContext.Provider>
+  );
+};
+
+export const processAuth = async (auth,setUser) => {
+
+  if (!auth){
+    return setUser({});
+  }
+
+  let firestoreResult = await firestore
+    .doc("users/" + auth.uid)
+    .get();
+  if (typeof(firestoreResult.data()) === "undefined"){
+    await firestore.collection("users").doc(auth.uid).set({
+      uid: auth.uid,
+      fullname: auth.displayName,
+      age: '0',
```

Comment:
why is their age 0

Reply:
when we request data from Facebook, the app must be audited and verified by Facebook to obtain this information. that's why when a user register using Facebook, the age will be 0 and then they can edit it later.